src/pipecat/services/anthropic.py

```diff
@@ -94,7 +99,7 @@ def __init__(
         **kwargs,
     ):
         super().__init__(**kwargs)
-        self._client = AsyncAnthropic(api_key=api_key)
+        self._client = kwargs.pop('client', None) or AsyncAnthropic(api_key=api_key) # if the client is provided, use it and remove it, otherwise create a new one
```

Comment:
I think we could just pass a new `client` keyword argument directly and then do:

```
self._client = client or AsyncAnthropic(api_key=api_key)
```

so:

```

    def __init__(
        self,
        *,
        api_key: str,
        model: str = "claude-3-5-sonnet-20240620",
        params: InputParams = InputParams(),
        client = None,
        **kwargs,
    ):
```